### PR TITLE
Add rule to ensure unique TLS app names

### PR DESCRIPTION
**Pull request by matthewhughes-uw**
Duplicate names can cause all sorts of issues, see tickets for details

Ticket: DENA-991

### CODE REVIEW

**Comment by sbuliarca** [comment]
We should add this to the list of rules in main to be considere

File: rules/msk_unique_app_names.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("getting modules: %w", err)
+	}
+
+	var TLSAppModules hclext.Blocks
+	for _, moduleBlock := range modules.Blocks {
+		if _, ok := moduleBlock.Body.Attributes[commonNameAttribute]; ok {
```
Comment (sbuliarca) [inline rules/msk_unique_app_names.go:75]: This is a good point to only select the modules with the cert_common_name.

Wondering if it wouldn't be better to look at the source attribute and to get the ones with the tls-app suffix.

Also, we should do this for the msk_app_topics too.

Comment (sbuliarca) [reply rules/msk_unique_app_names.go:75]: On another thought I understand why you took only the ones with cert_common_name. It is a required attribute and it is validated by the tflint validate plugin. 

So I think we should actually do both in the end.. filter only for modules with tls-app source and take only the ones with cert_common_name.

Comment (matthewhughes-uw) [reply rules/msk_unique_app_names.go:75]: > This is a good point to only select the modules with the cert_common_name.
> 
> Wondering if it wouldn't be better to look at the source attribute and to get the ones with the tls-app suffix.
> 
> Also, we should do this for the msk_app_topics too.

My approach has just been to filter based on the existence of the key I'm linting (e.g. `{consume,produce}_topic` in the `msk_app_topics` case) what's the value in additionally checking on the `tls-app` suffix? Unless we think we'll run into naming conflicts?

Comment (sbuliarca) [reply rules/msk_unique_app_names.go:75]: I'm thinking that maybe in the future we'll have more than one module with the cert_common_name (maybe the quota module?) and in this way we'll make the targeting specific.
We can also leave it like this for now and handle it later, but I think the change should be quite easy to implement

Comment (matthewhughes-uw) [reply rules/msk_unique_app_names.go:75]: >  We can also leave it like this for now and handle it later, but I think the change should be quite easy to implement

I'm going to advocate for implementing this later: if/when we need it

**Pull request merged**